From 060aaafdc9d2d45a114ec3bf0daa42ff3b54f347 Mon Sep 17 00:00:00 2001
From: Hyun Kwon <hyun.kwon@xilinx.com>
Date: Wed, 22 Jul 2020 10:08:34 -0700
Subject: [PATCH 1453/1851] drm: xlnx: zynqmp: fix -Wunused-but-set-variable
 warnings

commit 79e2368a6bd8a3036c40d7feb82ed7b12fe31851 from
https://github.com/Xilinx/linux-xlnx.git

Fix all warnings such as below.

drivers/gpu/drm/xlnx/zynqmp_dpsub.c: In function 'zynqmp_dpsub_remove':
drivers/gpu/drm/xlnx/zynqmp_dpsub.c:127:11: warning: variable 'ret' set but not used [-Wunused-but-set-variable]
...

Signed-off-by: Hyun Kwon <hyun.kwon@xilinx.com>
Reviewed-by: Dylan Yip <dylan.yip@xilinx.com>
State: pending
Signed-off-by: Yaliang Wang <Yaliang.Wang@windriver.com>
---
 drivers/gpu/drm/xlnx/zynqmp_disp.c  | 5 ++---
 drivers/gpu/drm/xlnx/zynqmp_dp.c    | 3 +--
 drivers/gpu/drm/xlnx/zynqmp_dpsub.c | 2 +-
 3 files changed, 4 insertions(+), 6 deletions(-)

diff --git a/drivers/gpu/drm/xlnx/zynqmp_disp.c b/drivers/gpu/drm/xlnx/zynqmp_disp.c
index 103079fcb340..057ce20f7b32 100644
--- a/drivers/gpu/drm/xlnx/zynqmp_disp.c
+++ b/drivers/gpu/drm/xlnx/zynqmp_disp.c
@@ -3071,10 +3071,9 @@ static void zynqmp_disp_create_crtc(struct zynqmp_disp *disp)
 {
 	struct drm_plane *plane = &disp->layers[ZYNQMP_DISP_LAYER_GFX].plane;
 	struct drm_mode_object *obj = &disp->xlnx_crtc.crtc.base;
-	int ret;
 
-	ret = drm_crtc_init_with_planes(disp->drm, &disp->xlnx_crtc.crtc, plane,
-					NULL, &zynqmp_disp_crtc_funcs, NULL);
+	drm_crtc_init_with_planes(disp->drm, &disp->xlnx_crtc.crtc, plane,
+				  NULL, &zynqmp_disp_crtc_funcs, NULL);
 	drm_crtc_helper_add(&disp->xlnx_crtc.crtc,
 			    &zynqmp_disp_crtc_helper_funcs);
 	drm_object_attach_property(obj, disp->color_prop, 0);
diff --git a/drivers/gpu/drm/xlnx/zynqmp_dp.c b/drivers/gpu/drm/xlnx/zynqmp_dp.c
index eebf7e6a9d0e..9f08b669195e 100644
--- a/drivers/gpu/drm/xlnx/zynqmp_dp.c
+++ b/drivers/gpu/drm/xlnx/zynqmp_dp.c
@@ -1462,7 +1462,6 @@ zynqmp_dp_connector_atomic_set_property(struct drm_connector *connector,
 					uint64_t val)
 {
 	struct zynqmp_dp *dp = connector_to_dp(connector);
-	int ret;
 
 	if (property == dp->sync_prop) {
 		zynqmp_dp_set_sync_mode(dp, val);
@@ -1473,7 +1472,7 @@ zynqmp_dp_connector_atomic_set_property(struct drm_connector *connector,
 		if (bpc) {
 			drm_object_property_set_value(&connector->base,
 						      property, bpc);
-			ret = -EINVAL;
+			return -EINVAL;
 		}
 	} else {
 		return -EINVAL;
diff --git a/drivers/gpu/drm/xlnx/zynqmp_dpsub.c b/drivers/gpu/drm/xlnx/zynqmp_dpsub.c
index 9b3545348f7b..369d111b093c 100644
--- a/drivers/gpu/drm/xlnx/zynqmp_dpsub.c
+++ b/drivers/gpu/drm/xlnx/zynqmp_dpsub.c
@@ -141,7 +141,7 @@ static int zynqmp_dpsub_remove(struct platform_device *pdev)
 
 	pm_runtime_disable(&pdev->dev);
 
-	return err;
+	return ret;
 }
 
 static int __maybe_unused zynqmp_dpsub_pm_suspend(struct device *dev)
-- 
2.31.1

